From e8fa1db62b991ecdc029c3daaf429efb95d5d778 Mon Sep 17 00:00:00 2001
From: Tien Nguyen <tien.nguyen.uh@renesas.com>
Date: Fri, 22 May 2026 07:58:19 +0000
Subject: [PATCH 2/3] drpai: update remove callbacks for Linux 6.18

Linux 6.18 expects platform driver remove callbacks to return void.

Update the DRP-AI remove callbacks to match the new prototype by
changing the return type from int to void and dropping the unused return
values.

No functional change.

Upstream-Status: Pending

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>
---
 drivers/drpai/drpai-if.c     | 6 ++----
 drivers/drpai/drpai-if_v2h.c | 6 ++----
 2 files changed, 4 insertions(+), 8 deletions(-)

diff --git a/drivers/drpai/drpai-if.c b/drivers/drpai/drpai-if.c
index 25f8d09a11bd..8e26f4f6bfb7 100644
--- a/drivers/drpai/drpai-if.c
+++ b/drivers/drpai/drpai-if.c
@@ -151,7 +151,7 @@
 
 /* A function called from the kernel */
 static int drpai_probe(struct platform_device *pdev);
-static int drpai_remove(struct platform_device *pdev);
+static void drpai_remove(struct platform_device *pdev);
 static int drpai_open(struct inode *inode, struct file *file);
 static int drpai_close(struct inode *inode, struct file *file);
 static int drpai_flush(struct file *file, fl_owner_t id);
@@ -353,7 +353,7 @@ static int drpai_probe(struct platform_device *pdev)
     return 0;
 }
 
-static int drpai_remove(struct platform_device *pdev)
+static void drpai_remove(struct platform_device *pdev)
 {
     DRPAI_DEBUG_PRINT("start.\n");
 
@@ -361,8 +361,6 @@ static int drpai_remove(struct platform_device *pdev)
     drpai_unregist_device();
 
     DRPAI_DEBUG_PRINT("end.\n");
-
-    return 0;
 }
 
 static int drpai_open(struct inode *inode, struct file *file)
diff --git a/drivers/drpai/drpai-if_v2h.c b/drivers/drpai/drpai-if_v2h.c
index 0f18935c3ea4..547012139711 100644
--- a/drivers/drpai/drpai-if_v2h.c
+++ b/drivers/drpai/drpai-if_v2h.c
@@ -178,7 +178,7 @@
 
 /* A function called from the kernel */
 static int drpai_probe(struct platform_device *pdev);
-static int drpai_remove(struct platform_device *pdev);
+static void drpai_remove(struct platform_device *pdev);
 static int drpai_open(struct inode *inode, struct file *file);
 static int drpai_close(struct inode *inode, struct file *file);
 static int drpai_flush(struct file *file, fl_owner_t id);
@@ -415,7 +415,7 @@ static int drpai_probe(struct platform_device *pdev)
     return ret;
 }
 
-static int drpai_remove(struct platform_device *pdev)
+static void drpai_remove(struct platform_device *pdev)
 {
     DRPAI_DEBUG_PRINT("start.\n");
 
@@ -423,8 +423,6 @@ static int drpai_remove(struct platform_device *pdev)
     drpai_unregist_device();
 
     DRPAI_DEBUG_PRINT("end.\n");
-
-    return 0;
 }
 
 static int drpai_open(struct inode *inode, struct file *file)
-- 
2.43.0

